tasks/initialise.yml

```diff
@@ -25,17 +26,16 @@
     - name: postgres | Check for presence of "Modified by ome postgresql ansible role" in file
       ansible.builtin.lineinfile:
         path: "{{ postgresql_dist_confdir }}/postgresql.conf"
-        line: "#Modified by ome postgresql ansible role"
+        line: "#Ansible managed"
```

Comment:
Assuming the new logic is to always recreate a configuration file from the combination of the original PSQL configuration file and the role variables (in the same spirit as the previous `template` task), my impression is this check is redundant and the `config_file_changed` logic should be removed?

Reply:
The modified file check should work for the fresh installation, as the original configure file will be copied from the configuration file.

Reply:
I have changed the template to include  #{{ ansible_managed }}.

It will check for it, i.e. looking for "`#Ansible managed`" in the file; if it is not found it will mark it as an alerted configuration file and start looking for the org file (postgresql.conf.org) that may exist in the current folder,  if not it will invoke the tasks inside org_config_file.yml to determine the sample file.
If the configuration file does not have the   "`#Ansible managed`" it will assume the file is the original configuration file then copy it to postgresql.conf.org and use its contents in the configuration template.

The configuration file does not have "#Ansible managed" on `test122b`; it will assume it is the original configuration file and copy it to org.
I think because you are in check mode, the file is not copied so it failed when trying to get the file contents.

Reply:
Taking the example of `test122b` here

```
[sbesson@test122b-database ~]$  sudo tail -n 20 /var/lib/pgsql/16/data/postgresql.conf


#------------------------------------------------------------------------------
# CUSTOMIZED OPTIONS
#------------------------------------------------------------------------------

# Add settings for extensions here

listen_addresses = '*'

shared_buffers = 8002MB


listen_addresses = '*'

effective_cache_size = 23880MB

shared_buffers = 7960MB

[sbesson@test122b-database ~]$ sudo head -n 2 /var/lib/pgsql/16/data/postgresql.conf
#Modified by ome postgresql ansible role
# -----------------------------
```

It does not have the `#Ansible managed` marker but it is also not the original configuration.
In that case, copying it as `postgresql.conf.org` is incorrect and also means that every invocation of the role will append the options at the bottom of the configuration file.